server/s3_delete_object_action.cc

```diff
@@ -130,8 +130,12 @@ void S3DeleteObjectAction::add_object_oid_to_probable_dead_oid_list() {
 
   // prepending a char depending on the size of the object (size based bucketing
   // of object)
-  S3CommonUtilities::size_based_bucketing_of_objects(
-      oid_str, object_metadata->get_content_length());
+  size_t object_size = object_metadata->get_content_length();
+  if (object_metadata->is_object_extended()) {
```

Comment:
should we add else block and read the content length there, instead of doing that in above line ?

Reply:
well, no need of else block. The logic is first get size of object using get_content_length(). This should be sufficient for 95% of time, and we don't have to execute code within 'if' block. Only when object is extended, which is rare, we have to read object size using 'get_primary_obj_size()'. Moreover, the code also looks readable.

Reply:
Actually, there is no much difference in the way you suggested here. So, any way can be done. My suggestion is let it be as is, .